pkg/local_object_storage/writecache/writecache.go

```diff
@@ -141,10 +141,12 @@ func (c *cache) Open(readOnly bool) error {
 	return c.initCounters()
 }
 
-// Init runs necessary services.
+// Init runs necessary services. No-op in read-only mode.
 func (c *cache) Init() error {
-	c.initFlushMarks()
-	c.runFlushLoop()
+	if !c.db.IsReadOnly() {
```

Comment:
runtime mode change is possible, have you checked that moving to `R/W` does start workers again?

Reply:
yep, checked: `SetMode` calls `openStore` too, that's why we rely on the DB state here (it would be possible to have a separate field, but I considered it redundant)

if u mean test, then there is no in-box approach to catch init actions. I may add optional callback which will be used in test only, do u think it's worth it?